### DISCUSSION TITLE
Compiling natives with musl, fails, and idea why?

### DISCUSSION THREAD

**Discussion by 11notes** [AsamK/signal-cli#565] [General]
I try to compile libsignal-client with musl on alpine linux, but I fail. I try to compile it with crt-static for musl `RUST_BACKTRACE=full RUSTFLAGS="-C target-feature=-crt-static" cargo build -p libsignal-jni --release` and I get:

Compiling aes-gcm-siv v0.1.0 (/tmp/signal-client/rust/aes-gcm-siv)
   Compiling libsignal-protocol-rust v0.1.0 (/tmp/signal-client/rust/protocol)
error: failed to run custom build command for libsignal-protocol-rust v0.1.0 (/tmp/signal-client/rust/protocol)

Caused by:
  process didnt exit successfully: /tmp/signal-client/target/release/build/libsignal-protocol-rust-36e484a66fb69cc5/build-script-build (exit code: 101)
  --- stderr
  thread main panicked at called Result::unwrap() on an Err value: Os { code: 2, kind: NotFound, message: No such file or directory }, rust/protocol/build.rs:13:52
  stack backtrace:
     0:     0x55580199edf0 - std::backtrace_rs::backtrace::libunwind::trace::h8c624fbae309582e
                                 at /rustc/1773f60ea5d42e86b8fdf78d2fc5221ead222bc1/library/std/src/../../backtrace/src/backtrace/libunwind.rs:100:5
     1:     0x55580199edf0 - std::backtrace_rs::backtrace::trace_unsynchronized::h6564b88f58da4961
                                 at /rustc/1773f60ea5d42e86b8fdf78d2fc5221ead222bc1/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
     2:     0x55580199edf0 - std::sys_common::backtrace::_print_fmt::hbe578678360ab0f2
                                 at /rustc/1773f60ea5d42e86b8fdf78d2fc5221ead222bc1/library/std/src/sys_common/backtrace.rs:67:5
     3:     0x55580199edf0 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h85886a5b1444f439
                                 at /rustc/1773f60ea5d42e86b8fdf78d2fc5221ead222bc1/library/std/src/sys_common/backtrace.rs:46:22
     4:     0x5558019bd67c - core::fmt::write::h0dd4368b249898df
                                 at /rustc/1773f60ea5d42e86b8fdf78d2fc5221ead222bc1/library/core/src/fmt/mod.rs:1078:17
     5:     0x55580199ba42 - std::io::Write::write_fmt::hc7578591889dd609
                                 at /rustc/1773f60ea5d42e86b8fdf78d2fc5221ead222bc1/library/std/src/io/mod.rs:1518:15
     6:     0x5558019a13c5 - std::sys_common::backtrace::_print::h7af5ef92ac4e699e
                                 at /rustc/1773f60ea5d42e86b8fdf78d2fc5221ead222bc1/library/std/src/sys_common/backtrace.rs:49:5
     7:     0x5558019a13c5 - std::sys_common::backtrace::print::h77e9ada3756be86b
                                 at /rustc/1773f60ea5d42e86b8fdf78d2fc5221ead222bc1/library/std/src/sys_common/backtrace.rs:36:9
     8:     0x5558019a13c5 - std::panicking::default_hook::{{closure}}::h254f831e0dbd3d22
                                 at /rustc/1773f60ea5d42e86b8fdf78d2fc5221ead222bc1/library/std/src/panicking.rs:208:50
     9:     0x5558019a1068 - std::panicking::default_hook::h445779ca93675b1f
                                 at /rustc/1773f60ea5d42e86b8fdf78d2fc5221ead222bc1/library/std/src/panicking.rs:227:9
    10:     0x5558019a1b01 - std::panicking::rust_panic_with_hook::h843384feb0a8a96b
                                 at /rustc/1773f60ea5d42e86b8fdf78d2fc5221ead222bc1/library/std/src/panicking.rs:593:17
    11:     0x5558019a1647 - std::panicking::begin_panic_handler::{{closure}}::hfaf517513c6e3ebc
                                 at /rustc/1773f60ea5d42e86b8fdf78d2fc5221ead222bc1/library/std/src/panicking.rs:499:13
    12:     0x55580199f2ac - std::sys_common::backtrace::__rust_end_short_backtrace::haa362f88a9fd70f7
                                 at /rustc/1773f60ea5d42e86b8fdf78d2fc5221ead222bc1/library/std/src/sys_common/backtrace.rs:141:18
    13:     0x5558019a15a9 - rust_begin_unwind
                                 at /rustc/1773f60ea5d42e86b8fdf78d2fc5221ead222bc1/library/std/src/panicking.rs:495:5
    14:     0x5558019bca21 - core::panicking::panic_fmt::h5f46bd9f58c47694
                                 at /rustc/1773f60ea5d42e86b8fdf78d2fc5221ead222bc1/library/core/src/panicking.rs:92:14
    15:     0x5558019bc843 - core::option::expect_none_failed::h66cda1afe5499797
                                 at /rustc/1773f60ea5d42e86b8fdf78d2fc5221ead222bc1/library/core/src/option.rs:1268:5
    16:     0x55580187b3c0 - core::result::Result<T,E>::unwrap::h77c048fc6b2be014
    17:     0x5558018a7b64 - build_script_build::main::h259fc6eea4c3ce81
    18:     0x555801887a83 - core::ops::function::FnOnce::call_once::h9712d04950b3fc83
    19:     0x555801889619 - std::sys_common::backtrace::__rust_begin_short_backtrace::h1f686d8128bfb645
    20:     0x5558018876c9 - std::rt::lang_start::{{closure}}::hf9d04a7ae4a8577b
    21:     0x5558019a1e89 - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h779aef9928144a2e
                                 at /rustc/1773f60ea5d42e86b8fdf78d2fc5221ead222bc1/library/core/src/ops/function.rs:259:13
    22:     0x5558019a1e89 - std::panicking::try::do_call::h41ace32a5a1543ff
                                 at /rustc/1773f60ea5d42e86b8fdf78d2fc5221ead222bc1/library/std/src/panicking.rs:381:40
    23:     0x5558019a1e89 - std::panicking::try::h772486cdda60d5cf
                                 at /rustc/1773f60ea5d42e86b8fdf78d2fc5221ead222bc1/library/std/src/panicking.rs:345:19
    24:     0x5558019a1e89 - std::panic::catch_unwind::h4e44292a12fffde8
                                 at /rustc/1773f60ea5d42e86b8fdf78d2fc5221ead222bc1/library/std/src/panic.rs:396:14
    25:     0x5558019a1e89 - std::rt::lang_start_internal::h6b6c18b24684517f
                                 at /rustc/1773f60ea5d42e86b8fdf78d2fc5221ead222bc1/library/std/src/rt.rs:51:25
    26:     0x5558018876a8 - std::rt::lang_start::hb9e095886daafd4e
    27:     0x5558018a7c8b - main
    28:     0x7fb91102fa03 - <unknown>
warning: build failed, waiting for other jobs to finish...
error: build failed

**Comment by m-ueberall** [comment]
For the record, this has been reiterated in discussion #1453 (in short: fully static builds would require a static library `libsignal_jni.a` which currently does not exist).
